akka-stream/src/main/scala/akka/stream/javadsl/Tcp.scala

```diff
@@ -283,11 +283,10 @@ class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
       options: JIterable[SocketOption],
       halfClose: Boolean,
       idleTimeout: Duration): Source[IncomingConnection, CompletionStage[ServerBinding]] =
-    Source.fromGraph(
-      delegate
-        .bindTls(interface, port, sslContext, negotiateNewSession, backlog, immutableSeq(options), idleTimeout)
-        .map(new IncomingConnection(_))
-        .mapMaterializedValue(_.map(new ServerBinding(_))(ec).toJava))
+    Source.fromGraph(delegate
+      .bindTls(interface, port, sslContext, negotiateNewSession, backlog, immutableSeq(options), halfClose, idleTimeout)
```

Comment:
Can you create an issue (and a FIXME close to the ApiMayChange here) that we should look into removing that `halfClose` parameter. It doesn't exist in scaladsl (and there it's always set to true). It's probably correct that it should always be true for TLS.

Therefore we should also not change the behavior here by passing on the parameter. Could become an issue if someone is using `false`.

Reply:
Created  #26689

Reply:
And reverted change